x-pack/plugins/ml/public/application/components/jobs_awaiting_node_warning/new_job_awaiting_node_shared/new_job_awaiting_node_shared.tsx

```diff
@@ -0,0 +1,150 @@
+    }
+  }, [jobIds]);
+
+  const checkCloudInfo = useCallback(async () => {
+    if (unassignedJobCount === 0) {
+      return;
+    }
+
+    try {
+      const resp = await ml.mlInfo();
+      const cloudId = resp.cloudId ?? null;
+      setCloudInfo({
+        isCloud: cloudId !== null,
+        cloudId,
+        deploymentId: cloudId === null ? null : extractDeploymentId(cloudId),
+      });
+    } catch (error) {
+      setCloudInfo(null);
+      // eslint-disable-next-line no-console
+      console.error('Could not determine cloud information', error);
+    }
+  }, [unassignedJobCount]);
+
+  useEffect(() => {
+    checkCloudInfo();
+  }, [unassignedJobCount]);
+
+  useEffect(() => {
+    checkNodes();
+  }, [jobIds]);
+
+  if (unassignedJobCount === 0) {
+    return null;
+  }
+
+  return (
+    <>
+      <EuiCallOut
+        title={
+          <FormattedMessage
+            id="xpack.ml.jobsAwaitingNodeWarningShared.title"
+            defaultMessage="Awaiting machine learning node"
+          />
+        }
+        color="primary"
+        iconType="iInCircle"
+      >
+        <div>
+          <FormattedMessage
+            id="xpack.ml.jobsAwaitingNodeWarningShared.noMLNodesAvailableDescription"
+            defaultMessage="There {jobCount, plural, one {is} other {are}} {jobCount, plural, one {# job} other {# jobs}} waiting for machine learning nodes to start."
+            values={{
+              jobCount: unassignedJobCount,
+            }}
+          />
+          <EuiSpacer size="s" />
+          {cloudInfo &&
+            (cloudInfo.isCloud ? (
+              <>
+                <FormattedMessage
+                  id="xpack.ml.jobsAwaitingNodeWarningShared.isCloud"
+                  defaultMessage="Elastic Cloud will autoscale to add more ML capacity. This may take 5-20 minutes. "
+                />
+                {cloudInfo.deploymentId === null ? null : (
+                  <FormattedMessage
+                    id="xpack.ml.jobsAwaitingNodeWarningShared.isCloud"
+                    defaultMessage="You can monitor progress in the {link}."
+                    values={{
+                      link: (
+                        <EuiLink
+                          href={`https://cloud.elastic.co/deployments?q=${cloudInfo.deploymentId}`}
+                        >
+                          <FormattedMessage
+                            id="xpack.ml.jobsAwaitingNodeWarningShared.linkToCloud.linkText"
+                            defaultMessage="Elastic Cloud admin console"
+                          />
+                        </EuiLink>
+                      ),
+                    }}
+                  />
+                )}
+              </>
+            ) : (
+              <FormattedMessage
+                id="xpack.ml.jobsAwaitingNodeWarningShared.notCloud"
+                defaultMessage="You are not running in Elastic Cloud. Your administrator is responsible for adding an ML node."
```

Comment:
Another option which highlights the autoscaling feature:
```suggestion
                defaultMessage="Only Elastic Cloud deployments can autoscale; you must add machine learning nodes."
```
If you want a link to help them figure out how to do that, I'd recommend https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-node.html#ml-node (which would have to be added to the doc link service, or else use the existing [elasticsearch.nodeRoles link](https://github.com/elastic/kibana/blob/03007d0150386709e49457b9f0da20d2724b7bc6/src/core/public/doc_links/doc_links_service.ts#L214)

Reply:
updated in https://github.com/elastic/kibana/pull/112264/commits/29e50eb96da70c6697c63356ed41018e3b9d239e